src/core/src/op/constant.cpp

```diff
@@ -108,14 +141,30 @@ Constant::Constant(bool memset_allocation, const element::Type& type, const Shap
 }
 
 void Constant::allocate_buffer(bool memset_allocation) {
+    // memset_allocation flag is to switch on initialization of objects in memory for element::string type
+    // and set memory to zero for numeric element types
     m_data = std::make_shared<AlignedBuffer>(mem_size(), host_alignment());
     if (memset_allocation) {
-        std::memset(m_data->get_ptr(), 0, m_data->size());
+        if (m_element_type == ov::element::string) {
+            // initialize std::string objects in memory
+            auto size = shape_size(m_shape);
+            auto string_ptr = static_cast<std::string*>(get_data_ptr_nc());
+            std::uninitialized_fill_n(string_ptr, size, std::string());
```

Comment:
should we explicitly call dtors for `std::string` during Constant destruction? Like we do for Tensors

I see that AlignedBuffer treats its data as `char *`, so it does not call ctors for stored objects

Reply:
Will fix it today asap.

Reply:
Fixed.